claims.go

```diff
@@ -12,9 +12,116 @@ type Claims interface {
 	Valid() error
 }
 
-// Structured version of Claims Section, as referenced at
-// https://tools.ietf.org/html/rfc7519#section-4.1
-// See examples for how to use this with your own claim types
+// RegisteredClaims are a structured version of the JWT Claims Set,
+// restricted to Registered Claim Names, as referenced at
+// https://datatracker.ietf.org/doc/html/rfc7519#section-4.1
+//
+// This type can be used on its own, but then additional private and
+// public claims embedded in the JWT will not be parsed. The typical usecase
+// therefore is to embedded this in a user-defined claim type.
+//
+// See examples for how to use this with your own claim types.
+type RegisteredClaims struct {
+	// the `iss` (Issuer) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1
+	Issuer string `json:"iss,omitempty"`
+
+	// the `sub (Subject) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2
+	Subject string `json:"sub,omitempty"`
+
+	// the `aud` (Audience) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3
+	Audience []string `json:"aud,omitempty"`
```

Comment:
Curious, have you considered this being a custom type that supported both `string` and `[]string`

> In the general case, the "aud" value is an array of case-
   sensitive strings, each containing a StringOrURI value.  In the
   special case when the JWT has one audience, the "aud" value MAY be a
   single case-sensitive string containing a StringOrURI value.

We could then implement custom unmarshalling to accommodate both. From the library perspective, it'd be an implementation detail, but to the caller we'd continue to expose `[]string`, for example:

json `string` or `[]string` -> go `[]string`

Or is `[]string` good enough for most use-cases, and if a user _must_ have it as a `string` they implement their own claims?

Reply:
> Curious, have you considered this being a custom type that supported both `string` and `[]string`

I have seen this in other libraries and I did not really see the use case for it to be honest.

> 
> > In the general case, the "aud" value is an array of case-
> > sensitive strings, each containing a StringOrURI value.  In the
> > special case when the JWT has one audience, the "aud" value MAY be a
> > single case-sensitive string containing a StringOrURI value.
> 
> We could then implement custom unmarshalling to accommodate both. From the library perspective, it'd be an implementation detail, but to the caller we'd continue to expose `[]string`, for example:
> 
> json `string` or `[]string` -> go `[]string`
> 
> Or is `[]string` good enough for most use-cases, and if a user _must_ have it as a `string` they implement their own claims?

As far as I know, even if the aud is just a string instead of an array, it should still be correctly deserialised into a []string, so not sure if there is a use case of having just a string.

Reply:
> As far as I know, even if the aud is just a string instead of an array, it should still be correctly deserialised into a []string, so not sure if there is a use case of having just a string.

If the incoming JWT "aud" claim is a string (and not an array of strings) deserialization will fail:

```
json: cannot unmarshal string into Go struct field RegisteredClaims.aud of type []string
```

But this is fine, disregard this as the goal of this `RegisteredClaims` is to support the most common use case.

I wish the RFC just omitted this "special case" altogether.

Reply:
I have added a custom type now, named `StringArray`. Not sure about the naming, anyone got any good ideas?

Reply:
Apologies for the delay.

Can you please add a net new test for parsing `RegisteredClaims`, specifically a test that uses a new JWT with multiple audience claims? Afaics none of the tests have JWTs with multiple audiences. In the `UnmarshalJSON` method might need to switch on `[]interface{}` and assert for `string`.. tests should help here.

Also, what will happen if a user of this package needs to _generate_ a token with "aud" as a `string` instead of `[]string`? I.e., they need to return `{ "aud": "example.com" }`. Is it okay to point users at creating their own custom claims, or should the library natively support (de)-serializing "aud" claim both ways?

Reply:
Sorry for not progressing on this, I was quite busy at work, I hopefully have some time during the coming weekend to finalise this PR.

Reply:
Of course, it's open source and we try to do what we can :) 

The library is in a stable state and much of what we're adding is incremental improvements, so it's natural it'll take time.